cmd/extract.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"nlp-tool/internal"
+
+	"github.com/spf13/cobra"
+)
+
+var extractCmd = &cobra.Command{
+	Use:   "extract",
+	Short: "Extract top entities from a given URL",
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var url string
+		if len(args) > 0 {
+			url = args[0]
+		} else {
+			var err error
+			url, err = internal.UrlInput()
+			if err != nil {
+				log.Fatalf("Invalid input: %s", err)
+			}
+		}
```

Comment:
You could invert this logic to avoid the `else` statement. It isn't wrong, just to simplify the things.